config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,210 @@
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Liveness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi 
+      readinessProbe:
+        failureThreshold: 2
+        periodSeconds: 10
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              # Check if the registered ray nodes count is the same as PIPELINE_PARALLEL_SIZE
+              [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]] &
+              p1=$!
+
+              # Check if huggingface server health
+              curl --silent --max-time 5 http://localhost:8080 &
+              p2=$!
+
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Readiness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi & # Run this check in the background
+              p3=$!
+
+              # Wait for all processes to complete
+              wait $p1 $p2 $p3
+              wait $p1 && wait $p2 && wait $p3
+      startupProbe:
+        failureThreshold: 40
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              ray status
+  volumes:
+    - name: shm
+      emptyDir:
+        medium: Memory
+        sizeLimit: 12Gi
+  workerSpec:
+    containers:
+      - name: worker-container
+        image: kserve/huggingfaceserver:latest
+        command: ["bash", "-c"]
+        args:
+          - |
+            SECONDS=0
+
+            while true; do              
+              if (( SECONDS <= 120 )); then
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379" > /dev/null 2>&1; then
+                  echo "GCS is ready."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Waiting for GCS to be ready."
+              else
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                  echo "GCS is ready. Any error messages above can be safely ignored."
```

Comment:
For the mentions of GCS, could we point users to any trouble shooting guide in case it never becomes ready?

Reply:
Oh never mind. I just saw that you have it below.

Reply:
yes